### PR TITLE
Menhir research

### PR DESCRIPTION
**Pull request by OptimShi**

### CODE REVIEW

**Review by DridsGhost — CHANGES_REQUESTED** [review]
Cara not present in Xara landblock
Will need to update March 2010 
ACE-World-16PY-Patches/Database/Patches/2010-03-LayOfTheLand/6 LandBlockExtendedData/934B.sql

--------

Will require quest entires written for missing quests.
Couldnt find quest MenhirResearchNPCCounter
Couldnt find quest MenhirReaserchRingXXXComplete

--------

Field doesnt tell you anything if you use it after already being stamped. Could be normal?

--------

#09	41.9N, 33.3E Holtburg Menhir Ring wasnt showing up on first load. I tinkered with Z height and updated the landblock via SQL and reloaded.
Might want to double check off of the appveyor artifact on your end its working first shot.

-------
Emote

I was able to immediately start the quest over after handing in the mana stone.
NPC should check MenhirResearchQuestCompleted on use.

This DirectedBroadcast is not needed. ACE will spit it out automatically
 - DirectBroadcast: You hand over 1 of your Enchanted Mana Stones.
 
 39287 Golden Key is not a valid weenie. Dont think we have any multi use golden keys in the database yet. Could use 44718 as reference.
 
 Probability: 1 is not needed.

--------

Facility Hub Landblock has a duplicate

Landblock already exists in
2010-01-ReforgingThePast/6 LandBlockExtendedData/SQL/8A02.sql

You have a duplicate in
2010-03-LayOfTheLand/6 LandBlockExtendedData/SQL/8A02.sql

File: Database/Patches/2017-01-EndOfRetail/6 LandBlockExtendedData/SQL/A9B3.sql [hunk 1]
```diff
@@ -1,4 +1,4 @@
-DELETE FROM `landblock_instance` WHERE `landblock` = 0xA9B3;
+DELETE FROM `landblock_instance` WHERE `guid` >= 0x7A9B3000 and `guid` <= 0x7A9B301D;
```
Comment (DridsGhost) [inline Database/Patches/2017-01-EndOfRetail/6 LandBlockExtendedData/SQL/A9B3.sql:1]: Is this needed?

Review current A9B3 landblocks in 16PY and patches.

File: Database/Patches/2010-03-LayOfTheLand/9 WeenieDefaults/Generic/80200 Menhir Ring One.sql [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+INSERT INTO `weenie_properties_string` (`object_Id`, `type`, `value`)
+VALUES (80200,   1, 'Menhir Ring One') /* Name */;
+
+INSERT INTO `weenie_properties_d_i_d` (`object_Id`, `type`, `value`)
+VALUES (80200,   1,   33556024) /* Setup */
+     , (80200,   3,  536871008) /* SoundTable */
+     , (80200,   8,  100667465) /* Icon */;
+INSERT INTO `weenie_properties_emote` (`object_Id`, `category`, `probability`, `weenie_Class_Id`, `style`, `substyle`, `quest`, `vendor_Type`, `min_Health`, `max_Health`)
+VALUES (80200, 37 /* ReceiveLocalSignal */, 1, NULL, NULL, NULL, 'UseEnchantedManaStone', NULL, NULL, NULL);
```
Comment (DridsGhost) [inline Database/Patches/2010-03-LayOfTheLand/9 WeenieDefaults/Generic/80200 Menhir Ring One.sql:37]: Review non functional Emote setting. Dies at InqQuest

**Review by DridsGhost — CHANGES_REQUESTED** [review]

**Comment by OptimShi** [comment]
This should be ready to go!

Quest steps
* Talk to Cara the Apprentice
  - Checks if you have the stone (quest in progress) and are level 50+, and if your quest timer is up
  - Can also giver her Coalesced Mana (WCID 42516, 42517, 42518)
* Travel to Rings one through Five-Twenty, and turn in.
* Using the stone twice at a specific ring will give you a message (made up since no record, or possibly did not even happen in retail, in line with quest text)

* Note that there are three Ring Fifteen, one in each Society.
* The inert ring locations were added based on one instance in pcaps and discussion on Wiki Talk page https://asheron.fandom.com/wiki/Talk:Menhir_Research

* Menhir Mana Fields were added in numerous places, since the Wiki Talk and other details implied every location had a mana ganerating field.

* All Menhir Rings are exact duplicates of Ring One, with relevant text/names/wcids changed.


Cara the Apprentice
@teleloc 0x934b012e [33.52 27.812 17.205] -0.275855 0 0 0.961199

Menhir Ring One
@teleloc 0x33460007 [9.9915 159.784 43.5763] -0.95583 0 0 0.293919

Menhir Ring Two
@teleloc 0x2545002e [130.006 129.677 -0.063] 1 0 0 0

Menhir Ring Three
@teleloc 0xc9900018 [65.3247 180.268 14.875] 1 0 0 0

Menhir Ring Four
@teleloc 0xc02d0036 [159.961 129.851 258.005] 1 0 0 0

Menhir Ring Five
@teleloc 0xbc11011f [60 85.4514 240] 1 0 0 0

Menhir Ring Six
@teleloc 0xb5710000 [184.957 80.0807 61.4501] -0.831272 0 0 -0.555866

Menhir Ring Seven
@teleloc 0xb1280023 [109.907 50.0235 170.71] 1 0 0 0

Menhir Ring Eight
@teleloc 0xaacc0000 [9.95342 100.146 48.005] 1 0 0 0

Menhir Ring Nine
@teleloc 0xa9b30007 [11.5 157.581 94.5] 0.00978612 0 0 -0.999952

Menhir Ring Ten
@teleloc 0x95970000 [117.926 109.011 224.005] -0.749927 0 0 0.661521

Menhir Ring Eleven
@teleloc 0x86890000 [26.9817 126.69 106.811] 1 0 0 0

Menhir Ring Twelve
@teleloc 0x81930000 [123.916 88.3596 124.005] 0.0341096 0 0 0.999418

Menhir Ring Thirteen
@teleloc 0x3d032f [179.666 -39.9263 -11.995] 1 0 0 0

Menhir Ring Fourteen
@teleloc 0x3a032f [179.666 -39.9263 -11.995] 1 0 0 0

Menhir Ring Fifteen
@teleloc 0xb7011e [110.072 -140.251 -23.5] 1 0 0 0
@teleloc 0xb8011e [110.072 -140.251 -23.5] 1 0 0 0
@teleloc 0xb9011e [109.972 -140.251 -22.5] 1 0 0 0

Menhir Ring Sixteen
@teleloc 0xb5f00121 [115.431 -60.4282 -81.195] 1 0 0 0

Menhir Ring Seventeen
@teleloc 0xa8bf0000 [36 156 86.01] 1 0 0 0

Menhir Ring Eighteen
@teleloc 0x99e6002f [128.811 159.962 78.5] 1 0 0 0

Menhir Ring Nineteen
@teleloc 0x866d0000 [9.58194 188.941 -0.445] 1 0 0 0

Menhir Ring Twenty
@teleloc 0x22bc0000 [36 180 160.005] 1 0 0 0

Inert Menhir Ring
@teleloc 0xb70100 [60 -60 -29.125] 0.673521 0 0 0.739168
@teleloc 0xb70101 [70 -60 -29.125] 0.727014 0 0 0.686623
@teleloc 0xb70102 [80 -60 -29.125] 0.727014 0 0 0.686623
@teleloc 0xb80100 [60 -60 -29.125] 0.673521 0 0 0.739168
@teleloc 0xb80101 [70 -60 -29.125] 0.727014 0 0 0.686623
@teleloc 0xb80102 [80 -60 -29.125] 0.727014 0 0 0.686623
@teleloc 0xb90100 [60 -60 -29.125] 0.673521 0 0 0.739168
@teleloc 0xb90101 [70 -60 -29.125] 0.727014 0 0 0.686623
@teleloc 0xb90102 [80 -60 -29.125] 0.727014 0 0 0.686623
@teleloc 0x1ab70013 [60.0403 59.5566 96.875] 0.921061 0 0 0.389418
@teleloc 0x1abe0039 [180.273 13.6487 148.875] -0.495261 0 0 0.868744
@teleloc 0x20b90025 [108.04 108.103 98.875] 0.120503 0 0 0.992713
@teleloc 0x23c10036 [155.831 131.377 110.875] 0.930508 0 0 -0.366273
@teleloc 0x8b64000a [34.6988 35.8856 10.993] 1 0 0 0
@teleloc 0xb93d0035 [155.902 108.042 54.875] 0.00638827 0 0 0.99998

File: Database/Patches/2010-03-LayOfTheLand/8 QuestDefDB/MagnificentBraceletPickupTimer.sql [hunk 3]
```diff
@@ -0,0 +1,4 @@
+DELETE FROM quest WHERE name = 'MagnificentBraceletPickupTimer';
```
[the remark this answers is not in the document]
Comment (OptimShi) [reply Database/Patches/2010-03-LayOfTheLand/8 QuestDefDB/MagnificentBraceletPickupTimer.sql]: This appears to user-error...probably copied that to use a template and screwed things up. Nice catch, thanks!

File: Database/Patches/2017-01-EndOfRetail/6 LandBlockExtendedData/SQL/A9B3.sql (hunk 1 above)
Comment (OptimShi) [reply Database/Patches/2017-01-EndOfRetail/6 LandBlockExtendedData/SQL/A9B3.sql]: Yes, this change to only delete the specific GUIDs is needed.

File: Database/Patches/2010-03-LayOfTheLand/9 WeenieDefaults/Generic/80200 Menhir Ring One.sql (hunk 2 above)
Comment (OptimShi) [reply Database/Patches/2010-03-LayOfTheLand/9 WeenieDefaults/Generic/80200 Menhir Ring One.sql]: Was fixed.

**Pull request merged**